src/openrct2/management/Research.cpp

```diff
@@ -601,20 +601,29 @@ void ride_entry_set_invented(int32_t rideEntryIndex)
 
 bool scenery_is_invented(const ScenerySelection& sceneryItem)
 {
-    assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT);
-    return _researchedSceneryItems[sceneryItem.SceneryType][sceneryItem.EntryIndex];
+    Guard::Assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT, "Invalid Scenery Type");
```

Comment:
Move the assert to the else as well and just have it assert `false`.

Reply:
Done.